array_api_compat/common/_linalg.py

```diff
@@ -8,7 +8,7 @@
 if np.__version__[0] == "2":
     from numpy.lib.array_utils import normalize_axis_tuple
 else:
-    from numpy.core.numeric import normalize_axis_tuple
+    from numpy.core.numeric import normalize_axis_tuple  # type: ignore[no-redef]
```

Comment:
Mypy's `allow_redefinition` will be enabled by default in mypy 2. So perhaps we could enable it project-wide? I believe that this ignore also wouldn't be necessary anymore.

Reply:
Adding `allow_redefinition = true` to pyproject.toml doesn't seem to have any effect in mypy 1.15.0